### PR TITLE
fix(web/server) uploaded asset in shared link not loaded

### LINKED ISSUE #1757: [BUG] Uploading photos through album public link does not show in album
### The bug

I tried letting others upload photos through the album public link. The uploads were successful and can be viewed under the timeline of the user who created the link but the uploaded photos does not show up in the  public album. 

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

v1.47.2

### Version of Immich Mobile App

v1.47

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    container_name: immich_server
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-server.sh"]
    volumes:
      - /mnt/sda/srv/immich/upload:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
      - TZ=Asia/Singapore
      - IMMICH_MACHINE_LEARNING_URL=false
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-microservices.sh"]
    volumes:
      - /mnt/sda/srv/immich/upload:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
      - IMMICH_MACHINE_LEARNING_URL=false
    depends_on:
      - redis
      - database
    restart: always

#  immich-machine-learning:
#    image: altran1502/immich-machine-learning:release
#    container_name: immich_machinelearning
#    entrypoint: ["/bin/sh", "./entrypoint.sh"]
#    volumes:
#      - /mnt/sda/srv/immich/upload:/usr/src/app/upload
    #env_file:
   #   - .env
  #  environment:
 #     - NODE_ENV=production
 #   depends_on:
#      - database
#    restart: always

  immich-web:
    image: altran1502/immich-web:release
    container_name: immich_web
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: redis:7
    restart: always

  database:
    container_name: immich_postgres
    image: postgres:15
    env_file:
      - .env
    environment:
    #  POSTGRES_PASSWORD: ${DB_PASSWORD}
    #  POSTGRES_USER: ${DB_USERNAME}
    #  POSTGRES_DB: ${DB_DATABASE_NAME}
      PG_DATA: /var/lib/postgresql/data
    volumes:
      - /srv/immich/pgdata:/var/lib/postgresql/data
    restart: always


  immich-proxy:
    container_name: immich_proxy
    image: altran1502/immich-proxy:release

    logging:
      driver: none
    depends_on:
      - immich-server
    restart: always
```

### Your .env content

```Shell
DB_HOSTNAME=immich_postgres
DB_USERNAME=
DB_PASSWORD=
DB_DATABASE_NAME=immich
REDIS_HOSTNAME=immich_redis
UPLOAD_LOCATION=/srv/immich/upload
JWT_SECRET=
ENABLE_MAPBOX=false
```

### Reproduction steps

```bash
1. Create album
2. Create link (public sharing)
3. Allow public upload
4. Let others upload photos using the link
5. Upload success but the photos won't show in the shared album
...
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alextran1502**
Fixes #1757 

* Create endpoint for add/remove asset in individual shared link
* Fix issue with assets added into shared album doesn't show up
* Convert callback to async/await

### CODE REVIEW

File: server/apps/immich/src/api-v1/asset/dto/add-assets-to-shared-link.dto.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { IsNotEmpty } from 'class-validator';
 
-export class UpdateAssetsToSharedLinkDto {
+export class AddAssetToSharedLinkDto {
   @IsNotEmpty()
-  assetIds!: string[];
+  ids!: string[];
```
Comment (jrasm91) [inline server/apps/immich/src/api-v1/asset/dto/add-assets-to-shared-link.dto.ts:5]: We should maybe consider using a single dto for adding assets in all our endpoints. They should ideally work the same way. Same with bulk remove assets. Basically here and adding to an album.

File: server/apps/immich/src/api-v1/asset/dto/remove-assets-from-shared-link.dto.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { IsNotEmpty } from 'class-validator';
+
+export class RemoveAssetsFromSharedLinkDto {
+  @IsNotEmpty()
+  ids!: string[];
```
Comment (jrasm91) [inline server/apps/immich/src/api-v1/asset/dto/remove-assets-from-shared-link.dto.ts:5]: Same as above.

File: server/libs/domain/src/share/share.core.ts [hunk 3]
```diff
@@ -63,13 +63,24 @@ export class ShareCore {
...
 
-  async updateAssets(userId: string, id: string, assets: AssetEntity[]) {
+  async addAssets(userId: string, id: string, assets: AssetEntity[]) {
     const link = await this.get(userId, id);
     if (!link) {
       throw new BadRequestException('Shared link not found');
     }
 
-    return this.repository.save({ ...link, assets });
+    return this.repository.save({ ...link, assets: [...link.assets, ...assets] });
```
Comment (jrasm91) [inline server/libs/domain/src/share/share.core.ts:72]: Will this cause an error if an asset already had been added to the shared link? I assume there is a unique constant they will be violated. How is that handled in the add to album flow?

**Review by jrasm91 — APPROVED** [review]

File: server/apps/immich/src/api-v1/asset/dto/add-assets-to-shared-link.dto.ts (hunk 1 above)
Comment (jrasm91) [reply server/apps/immich/src/api-v1/asset/dto/add-assets-to-shared-link.dto.ts:5]: No, I mean that this seems really similar to adding and removing assets from an _album_. In fact, the DTOs are basically exactly the same for adding to a shared link as to adding to an album. Can we use a single "AddAssetsDto" in both places?

Comment (alextran1502) [reply server/apps/immich/src/api-v1/asset/dto/add-assets-to-shared-link.dto.ts:5]: Got it, sounds good!

File: server/libs/domain/src/share/share.core.ts (hunk 3 above)
Comment (alextran1502) [reply server/libs/domain/src/share/share.core.ts:72]: They just got added to the album and the shared link. There is no error regarding the unique constraint.

**Pull request merged**